### ISSUE TITLE
Popup "You have been idle for.." is showing while in a break

### ISSUE THREAD

**Issue by nkrot** [johannesjo/super-productivity#1834]
When I am taking a longer break, like 20 minutes after a number of Pomodoro sessions, the application is still throwing a pop up "You have been idle for ....". This is annoying and contradicting the idea of a longer break, isn't it?

### Your Environment

- Version used: 7.9.1 (deb)
- Operating System and version: Ubuntu 18.04
- Desktop Environment: MATE
- Browser Name and version: Firefox 89.0

### Expected Behavior

When I am in a longer break, Idle Time popup should not be shown. It may be shown once longer break has elapsed AND idle time length has also elapsed. For example, if my longer break period is set 20 minutes and Idle time is set to 5 minutes, then the popup may appear once 25 minutes elapsed.

**Comment by johannesjo** [comment]
Hey there! Thanks for opening this up. I think this is not a 100% clear since the idle time might have been used for something else than a break, e.g. a meeting or talking to a colleague or something, so I think it makes most sense to leave the decision what to do with the idle time to the user.

But I can understand why you would prefer it to be different. PRs for implementing this as opt in optional feature are welcome!

**Comment by nkrot** [comment]
@johannesjo , i think i did not make myself clear. The application *knows* that I am taking a break, it is showing me a countdown timer for the break. I am referring to the case that the Idle pop-up does not make sense while break timer is running.

If the user is on a meeting, he or she should have skipped the "take a break" reminder.

**Comment by johannesjo** [comment]
> If the user is on a meeting, he or she should have canceled the "take a break" reminder.

He maybe should, but it doesn't mean he always does. Think of the following case: User is in work session and a colleague asks him something. The (work related) discussion takes much longer than the pomodoro session and the break already started. What you'd expect? Or you're outside for your break and a colleague asks you something. 

These are cases (and there are probably more) where I personally would prefer to have an Idle Dialog shown.

**Comment by nkrot** [comment]
the scenario that you describe does not sound convincing to me when it comes to showing idle dialogue *during* a break. the dialogue can be show *after* the break ended. logically, users are supposed to be idle while taking a break.

> The (work related) discussion takes much longer than the pomodoro session and the break already started. What you'd expect?

I think this was a topic of another issue that relates to *manually confirming* a start of a pomodoro session or of a break. Until confirmation event was received, the current session or break continues. and when the break ends (+plus some time) and there is still no input from the user, then idle dialog can appear: logically, a user is not supposed to be idle during supposedly a working session :)

Anyway, personally i feel very annoyed by modern software trend to overwhelm users with excessive notifications. Every app that steals a second of my life because it wants me to  react to unnecessarily annoying notification will vanish. Just a side note :)

**Comment by johannesjo** [comment]
> the scenario that you describe does not sound convincing to me when it comes to showing idle dialogue _during_ a break. the dialogue can be show _after_ the break ended. logically, users are supposed to be idle while taking a break.

Now, showing the idle dialog only after is a bit of a different thing. Makes sense to me to implement it like this (the idle dialog does not appear during the break, but after it being finished and appears when minIdleTime > idleTime - breakTime with idleTime - breakTime as value) . Furthermore as I already said, we can implement not triggering the idle dialog at all during pomodoro breaks. PRs are welcome!
